pangeo_forge_recipes/rechunking.py

```diff
@@ -238,3 +239,45 @@ def _sort_by_speed_of_varying(item):
     ds_combined = xr.combine_nested(dsets_to_concat, concat_dim=concat_dims_sorted)
 
     return first_index, ds_combined
+
+
+def _gather_coordinate_dimensions(group: zarr.Group) -> List[str]:
+    return list(
+        set(itertools.chain(*(group[var].attrs.get("_ARRAY_DIMENSIONS", []) for var in group)))
+    )
+
+
+def consolidate_dimension_coordinates(
+    singleton_target_store: zarr.storage.FSStore,
+) -> zarr.storage.FSStore:
+    """Consolidate dimension coordinates chunking"""
+    group = zarr.open_group(singleton_target_store)
+
+    dims = (dim for dim in _gather_coordinate_dimensions(group) if dim in group)
+    for dim in dims:
+        arr = group[dim]
+        attrs = dict(arr.attrs)
+        data = arr[:]
+
+        # This will generally use bulk-delete API calls
+        # config.storage_config.target.rm(dim, recursive=True)
+
+        singleton_target_store.fs.rm(singleton_target_store.path + "/" + dim, recursive=True)
+
+        new = group.array(
+            dim,
+            data,
+            chunks=arr.shape,
+            dtype=arr.dtype,
+            compressor=arr.compressor,
+            fill_value=arr.fill_value,
+            order=arr.order,
+            filters=arr.filters,
+            overwrite=True,
+        )
+
+        new.attrs.update(attrs)
+        # Open question: Should this always be called
+        # or should we have the option of ConsolidateMetadata
+        zarr.consolidate_metadata(singleton_target_store)
```

Comment:
This was the fix @jbusecke and I came up with yesterday to get around the errors reported in the cmip6 pipeline. I think when the coordinate_dims are consolidated in this PR, the metadata has to be updated after. @jbusecke  might have a better way to explain this though!

Before the consolidated_metadata PR option was introduced all of the `StoreToZarr` recipes were consolidating metadata by default.
_from the docs_ ```The__ default (consolidated=None) means write consolidated metadata and attempt to read consolidated metadata for existing stores (falling back to non-consolidated).```

Reply:
Yep, that makes total sense @norlandrhagen!

Apologies the review here is a little disjointed. Down in https://github.com/pangeo-forge/pangeo-forge-recipes/pull/671#discussion_r1473536255 I mention my proposed fix for this, which is that we should _also remove_ the consolidate_metadata flag from StoreToZarr, and _always default_ `consolidated=False` in StoreToZarr.

Then it is up to the user if they want to do:

```
StoreToZarr() | ConsolidateCoordinateDimensions()
```

or

```
StoreToZarr() | ConsolidateCoordinateDimensions() | ConsolidateMetadata()
```

(☝️ seems like this is what @jbusecke would want to do)

or

```
StoreToZarr() | ConsolidateMetadata()
```

But in order to make these options possible, we need to start from an unconsolidated store emitted by `StoreToZarr()`, because as you observe here, if StoreToZarr emits a consolidated store, and coordinate dims are subsequently consolidated without then re-consolidating metadata, we end up in a broken state.

Always emitting an unconsolidated store from `StoreToZarr()` should resolve this.

Does that make sense?

Reply:
No worries!

It does fix it! However, then we are defaulting to non_consolidated coords, which seems like the wrong approach for most recipes. I think this is fine if everyone knows that they probably should add the ConsolidatedCoords transform into their recipe, but I could see it commonly slipping through the cracks. Maybe this is kind of what kind of guard rails / default assumptions we want to have 🤷 .

Reply:
I share the concern about this requirement for explicitly having to chain consolidation stages (in a specific order) leading to errors on the user side.  

Maybe we can consider renaming `StoreToZarr`-> `StoreToZarrUnconsolidated` (or something else) and subclass that as the new `StoreToZarr` (this would change the default behavior)

alternatively make a new class `StoreToZarrConsolidated` that subclasses the current `StoreToZarr`, this would not change the default behavior but probably would need a thorough rewrite of the docs to make sure users are using consolidated storage as a default?